bookkeeper-server/src/main/java/org/apache/bookkeeper/client/BookieWatcher.java

```diff
@@ -138,17 +142,40 @@ private synchronized void processReadOnlyBookiesChanged(Set<BookieSocketAddress>
      *
      * @throws BKException when failed to read bookies
      */
-    public void readBookiesBlocking() throws BKException {
-        this.registrationClient.watchReadOnlyBookies(bookies -> processReadOnlyBookiesChanged(bookies.getValue()));
-        this.registrationClient.watchWritableBookies(bookies -> processWritableBookiesChanged(bookies.getValue()));
+    public void initialBlockingBookieRead() throws BKException {
+        CompletableFuture<?> writable;
+        CompletableFuture<?> readonly;
+        synchronized (this) {
+            if (initialReadonlyBookiesFuture == null
+                && initialWritableBookiesFuture == null) {
```

Comment:
Is it possible one of these is non-null? If so we need to handle that, if not just check for one and assert on second.

Reply:
will change